worlds/stardew_valley/client.py

```diff
@@ -0,0 +1,250 @@
+            logger.warning("Internal logic was not able to load, check your yamls and relaunch.")
+            return
+
+        try:
+            rule = self.ctx.logic.region.can_reach_location(location)
+            state = get_updated_state(self.ctx)
+            simplified, _ = rule.evaluate_while_simplifying(state)
+            expl = explain(simplified, state, mode=ExplainMode.CLIENT)
+        except KeyError:
+
+            result, usable, response = Utils.get_intended_text(location, [loc.name for loc in self.ctx.multiworld.get_locations(1)])
+            if usable:
+                rule = self.ctx.logic.region.can_reach_location(result)
+                state = get_updated_state(self.ctx)
+                simplified, _ = rule.evaluate_while_simplifying(state)
+                expl = explain(simplified, state, mode=ExplainMode.CLIENT)
+            else:
+                logger.warning(response)
+                return
+
+        self.ctx.previous_explanation = expl
+        self.ctx.ui.print_json(parse_explanation(expl))
+
+    @mark_raw
+    def _cmd_more(self, index: str = ""):
+        """Will tell you what's missing to consider a location in logic."""
+        if self.ctx.previous_explanation is None:
+            logger.warning("No previous explanation found.")
+            return
+
+        try:
+            expl = self.ctx.previous_explanation.more(int(index))
+        except (ValueError, IndexError):
+            logger.info("Which previous rule do you want to explained?")
+            for i, rule in enumerate(self.ctx.previous_explanation.more_explanations):
+                logger.info(f"/more {i} -> {str(rule)})")
+            return
+
+        self.ctx.previous_explanation = expl
+        self.ctx.ui.print_json(parse_explanation(expl))
+
+    if not tracker_loaded:
+        del _cmd_explain
+        del _cmd_explain_missing
+
+
+class StardewClientContext(TrackerGameContext):
+    game = "Stardew Valley"
+    command_processor = StardewCommandProcessor
+    logic: StardewLogic | None = None
+    previous_explanation: RuleExplanation | None = None
+
+    def make_gui(self):
+        ui = super().make_gui()  # before the kivy imports so kvui gets loaded first
+
+        class StardewManager(ui):
+            base_title = f"Stardew Valley Tracker with UT {UT_VERSION} for AP version"  # core appends ap version so this works
+            ctx: StardewClientContext
+
+            def build(self):
+                container = super().build()
+                if not tracker_loaded:
+                    logger.info("To enable the tracker page, install Universal Tracker.")
+
+                return container
+
+        return StardewManager
+
+    def setup_logic(self):
+        if self.multiworld is not None:
+            self.logic = self.multiworld.worlds[1].logic
+
+
+def parse_explanation(explanation: RuleExplanation) -> list[JSONMessagePart]:
+    result_regex = r"(\(|\)| & | -> | \| |\d+x | \[|\](?: ->)?\s*| \(\w+\)|\n\s*)"
```

Comment:
Any chance this regex could get maybe a comment or some splitting down? Regex are famously unreadable after about 7 seconds after merging